### ISSUE TITLE
<remove /> nodes are not processed.

### ISSUE THREAD

**Issue by danslo** [danslo/hhvm#8]
When creating a configurable option, you're faced with the following:

Both the configurable attribute section and default attributes are shown.

**Comment by danslo** [comment]
This comes down to the following bug in remove nodes:

``` php
<?php

$xml = simplexml_load_string(<<<EOF
<root>
  <block name="test" />
</root>
EOF
);

$block = $xml->xpath("//block[@name='test']");
$block[0]->addAttribute('ignore', true);

var_dump($block[0]->attributes()->ignore);   // exists
var_dump($xml->block->attributes()->ignore); // doesn't exist
```

**Comment by danslo** [comment]
The problem is that xpath doesn't return original xml nodes, it simply creates them. 

We can do two things:
1) Implement recursive lookup for element->m_node against xpath xmlNodePtr. 
2) Create a map between xmlNodePtr and c_SimpleXMLElements. 

Not sure which is preferable.

**Comment by danslo** [comment]
Fixed with e1008edbb9cc393de09c6eadbe9cec7c4729b6df.
